WireAPI/Tests/WireAPITests/EventDecoding/UserEventDecodingTests.swift

```diff
@@ -0,0 +1,258 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+@testable import WireAPI
+import XCTest
+
+final class UserEventDecodingTests: XCTestCase {
+
+    private let helper = EventDecodingAssertionHelper()
+
+    func testDecodingClientAddEvent() throws {
+        try helper.assertEventDecodingFromResource(
+            named: "UserClientAdd",
+            to: .user(.clientAdd(Scaffolding.clientAddEvent))
+        )
+    }
```

Comment:
I see the helper as a nice way to avoid duplication. Though in unit tests I would like to quickly to understand what is loaded, tested and expected. 

Also dependencies between files/structs/classes in unit tests causes more issues once one starts to modularise and move tests around. Duplication is not bad for unit tests.

So I would prefer to suggest to use it a bit more "plain":
```swift
    func testDecodingClientAddEvent() throws {
        // Given event data.
        let resource = try MockEventDataResource(name: "UserClientAdd")

        // When decode update event.
        let updateEvent = try JSONDecoder.defaultDecoder.decode(UpdateEvent.self, from: resource.jsonData)

        // Then
        XCTAssertEqual(updateEvent, .user(.clientAdd(Scaffolding.clientAddEvent)))
    }
```

Reply:
maybe JSONDecoder.defaultDecoder can be a `var mockJSONDecoder` in the test case

Reply:
done e04207ee4b3759ed6fb5c52d78191a2c198fdda2